### DISCUSSION TITLE
Rich text upload > collection > field marked as required but not actually required

### DISCUSSION THREAD

**Discussion by ciokan** [payloadcms/payload#1090] [Roadmap]
# Bug Report

Consider the following block:

import type {Block} from 'payload/types';

```
const RichTextBlock: Block = {
	slug        : 'RichText', // required
	imageURL    : 'https://google.com/path/to/image.jpg',
	imageAltText: 'A nice thumbnail image to show what this block looks like',
	fields      : [ // required
		{
			name        : 'content', // required
			type        : 'richText', // required
			defaultValue: [{
				children: [{text: 'Here is some default content for this field'}],
			}],
			required    : true,
			admin       : {
				elements  : [
					'h2',
					'h3',
					'h4',
					'link',
					'upload',
				],
				leaves    : [
					"bold",
					"italic",
					"underline",
					"strikethrough",
					"code",
				],
				hideGutter: true,
				upload    : {
					collections: {
						media: {
							fields: [{
								name     : 'alt',
								label    : 'Alt text',
								type     : 'text',
								minLength: 25,
								maxLength: 255,
								required : true,
							},],
						},
					},
				},
			}
		}
	]
};


export default RichTextBlock;
```

## Expected Behavior

the `alt` field should be required

## Current Behavior

The alt field does not even show when adding media. Only if you click edit.

**Comment by denolfe** [comment]
Thanks for reporting this @ciokan. We agree that it would be ideal to validate the nested rich text upload fields on the parent document save. We definitely want to improve this workflow. We'll file this under a feature enhancement for now and get back to this as soon as possible.

**Comment by JarrodMFlesch** [comment]
If you stumble upon this thread, this was a very old example of Payload when using Slate, as of now we recommend you use the lexical RTE, where validation does work properly (among many many other superior things).

Side note: this thread is a time capsule for me, time really flies.
